core/app/models/spree/shipment.rb

```diff
@@ -4,6 +4,14 @@ module Spree
   # An order's planned shipments including tracking and cost.
   #
   class Shipment < Spree::Base
+    class InvalidStateChange < StandardError; end
+
+    READY = 'ready'
+    PENDING = 'pending'
+    SHIPPED = 'shipped'
+    CANCELED = 'canceled'
+    VALID_STATES = [READY, PENDING, SHIPPED, CANCELED]
```

Comment:
Can we rename this constant to `DEFAULT_STATES`? We don't want to give the impression that overwriting a constant is a good extension point (because it's not)